client/service/workspace_configrations.go

```diff
@@ -0,0 +1,34 @@
+package service
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// WorkspaceConfAPI exposes the workspace configurations API
+type WorkspaceConfAPI struct {
+	Client *DBApiClient
+}
+
+// Update will handle creation of new values as well as deletes. Deleting just implies that a value of "" is sent with
+// the appropriate key
+func (a WorkspaceConfAPI) Update(workspaceConfMap map[string]string) error {
+	_, err := a.Client.performQuery(http.MethodPatch, "/preview/workspace-conf", "2.0", nil, workspaceConfMap, nil)
+	return err
+}
+
+// Read just returns back a map of keys and values which keys are the configuration items and values are the settings
+func (a WorkspaceConfAPI) Read(keys string) (map[string]string, error) {
+	var wsConfResp map[string]string
```

Comment:
This may need a todo just to verify that the maps are map[string]string and not map[string]interface{} as json supports string -> string, string -> float64, string -> bool and string -> struct... I think it is string -> string but worth verifying and documenting that.

Reply:
I tested that.  If you pass anything but a string in for any of the keys you get: 
{
    "error_code": "BAD_REQUEST",
    "message": "Values must be strings"
}

Reply:
Adding some comments with a TODO marked.  The API returns an error for any value passed in that isn't of type string in json.